<h1 style="text-align: center;"><a title="Data Prep-AIMS-Cmr-2024-25">Assignment 1, Data Prep  @AIMSSen 25</h1>

**Instructor:** 

* Rockefeller

# Exercice 1:  Trends in Business in Nigeria

**NaijaBiz Analytics** is a leading business intelligence firm that specializes in analyzing small and medium enterprises (SMEs) across Nigeria. The company has been contracted by the Nigerian Ministry of Industry, Trade and Investment to conduct a comprehensive analysis of business performance across different states and sectors. 

As a newly hired Data Analyst at NaijaBiz Analytics, you have been provided with a dataset containing information about businesses across various Nigerian states. Your task is to analyze the business data using basic filtering operations to provide actionable insights.

**Dataset:** `naija_businesses.csv`

For each question below, provide the Python pandas code and extract meaningful business insights from your analysis.


In [3]:
!pip install pandas

  Using cached pandas-2.3.3-cp312-cp312-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl.metadata (91 kB)
  Using cached numpy-2.3.4-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl.metadata (62 kB)
  Using cached pytz-2025.2-py2.py3-none-any.whl.metadata (22 kB)
  Using cached tzdata-2025.2-py2.py3-none-any.whl.metadata (1.4 kB)
Using cached pandas-2.3.3-cp312-cp312-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl (12.4 MB)
Using cached numpy-2.3.4-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl (16.6 MB)
Using cached pytz-2025.2-py2.py3-none-any.whl (509 kB)
Using cached tzdata-2025.2-py2.py3-none-any.whl (347 kB)


In [8]:
import pandas as pd
from IPython.display import display

In [9]:
pd.__version__

'2.3.3'

In [10]:
Nigerian_businesses = pd.read_csv("data/naija_businesses.csv")

Nigerian_businesses

business_id business_name   state  business_type  start_date  \
0       NG_0001    Business_1   Benue         Retail  2023-02-01   
1       NG_0002    Business_2  Kaduna  Manufacturing  2023-11-17   
2       NG_0003    Business_3     Oyo     Technology  2020-09-02   
3       NG_0004    Business_4   Benue  Manufacturing  2023-04-14   
4       NG_0005    Business_5  Rivers     Technology  2022-11-28   
..          ...           ...     ...            ...         ...   
195     NG_0196  Business_196    Kano       Services  2023-09-27   
196     NG_0197  Business_197    Kano    Agriculture  2021-01-31   
197     NG_0198  Business_198  Kaduna    Agriculture  2021-11-29   
198     NG_0199  Business_199     Oyo    Agriculture  2022-11-08   
199     NG_0200  Business_200  Rivers         Retail  2020-06-25   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                   417741               406718          8        False   
1                   696804               256593         10         True   
2                   520713                87719         11        False   
3                   553527               389310          7        False   
4                   231537               -11869         10         True   
..                     ...                  ...        ...          ...   
195                 611128               415487          8         True   
196                 916895               349547          7        False   
197                 778180               430501          8        False   
198                  98486                78788          4        False   
199                 713491               339290         10        False   

     accepts_mobile_money  
0                    True  
1                   False  
2                    True  
3                    True  
4                   False  
..                    ...  
195                 False  
196                  True  
197                  True  
198                  True  
199                 False  

[200 rows x 10 columns]

In [11]:
Nigerian_businesses.shape

(200, 10)

In [12]:
Nigerian_businesses.dtypes

business_id              object
business_name            object
state                    object
business_type            object
start_date               object
monthly_revenue_naira     int64
monthly_costs_naira       int64
employees                 int64
has_website                bool
accepts_mobile_money       bool
dtype: object

### Q1: High Revenue Businesses
Filter and display all businesses that have monthly revenue greater than 600,000 Naira. How many such businesses exist?

**Business Context:** Identifying high-revenue businesses can help understand successful business models.


In [13]:
df_monthly_revenue_gt_600k = Nigerian_businesses[Nigerian_businesses["monthly_revenue_naira"] > 600000]

df_monthly_revenue_gt_600k.head()

business_id business_name   state  business_type  start_date  \
1      NG_0002    Business_2  Kaduna  Manufacturing  2023-11-17   
6      NG_0007    Business_7     Oyo       Services  2021-06-21   
7      NG_0008    Business_8     Oyo     Technology  2022-07-09   
11     NG_0012   Business_12   Benue       Services  2022-06-21   
12     NG_0013   Business_13  Rivers  Manufacturing  2020-02-05   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
1                  696804               256593         10         True   
6                  724708               207436         12        False   
7                  928641               111173         11        False   
11                 835058               436180          7        False   
12                 677796               298681          6        False   

    accepts_mobile_money  
1                  False  
6                   True  
7                  False  
11                  True  
12                  True

In [14]:
int(df_monthly_revenue_gt_600k["business_name"].count())

60

60 businesses have monthly revenue greater than 600,000 Naira.

### Q2: Technology Businesses with Websites
Filter businesses that are in the "Technology" sector AND have a website. Display their business names and monthly revenue.

**Business Context:** Understanding digital adoption in the tech sector can inform technology support programs.


In [15]:
Tech_businesses = Nigerian_businesses.loc[(Nigerian_businesses["business_type"] == "Technology") & (Nigerian_businesses["has_website"] == True)]

Tech_businesses

business_id business_name    state business_type  start_date  \
4       NG_0005    Business_5   Rivers    Technology  2022-11-28   
9       NG_0010   Business_10     Kano    Technology  2021-06-08   
33      NG_0034   Business_34     Kano    Technology  2021-07-23   
74      NG_0075   Business_75      Oyo    Technology  2022-12-17   
115     NG_0116  Business_116   Kaduna    Technology  2020-12-22   
144     NG_0145  Business_145  Anambra    Technology  2022-12-02   
161     NG_0162  Business_162    Lagos    Technology  2020-04-24   
176     NG_0177  Business_177   Rivers    Technology  2022-07-08   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
4                   231537               -11869         10         True   
9                   549862               293803          3         True   
33                  265604                 8259          8         True   
74                  370740               296592          6         True   
115                 742563               272479          8         True   
144                 349510               294444          8         True   
161                 319976               255980         12         True   
176                 673369               261477         10         True   

     accepts_mobile_money  
4                   False  
9                   False  
33                   True  
74                   True  
115                  True  
144                  True  
161                 False  
176                 False

In [16]:
Tech_businesses[["business_name", "monthly_revenue_naira"]]

business_name  monthly_revenue_naira
4      Business_5                 231537
9     Business_10                 549862
33    Business_34                 265604
74    Business_75                 370740
115  Business_116                 742563
144  Business_145                 349510
161  Business_162                 319976
176  Business_177                 673369

### Q3: Manufacturing Business in Specific States

Filter all Manufacturing businesses located in either "Lagos" OR "Kano" states. How many manufacturing businesses are in these two states?

**Business Context:** These states are major industrial hubs, and this analysis helps understand manuafacturing concentration.

In [20]:
in_Lagos_or_Kano = Nigerian_businesses[(Nigerian_businesses["state"] == "Lagos") | (Nigerian_businesses["state"] == "Kano")]

in_Lagos_or_Kano.head()

business_id business_name  state  business_type  start_date  \
9      NG_0010   Business_10   Kano     Technology  2021-06-08   
22     NG_0023   Business_23   Kano  Manufacturing  2023-01-20   
27     NG_0028   Business_28   Kano         Retail  2020-02-20   
31     NG_0032   Business_32  Lagos         Retail  2020-05-29   
33     NG_0034   Business_34   Kano     Technology  2021-07-23   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
9                  549862               293803          3         True   
22                 589101               332379         11        False   
27                 713895               168645         10        False   
31                 324106               259284          3        False   
33                 265604                 8259          8         True   

    accepts_mobile_money  
9                  False  
22                 False  
27                 False  
31                  True  
33                  True

In [24]:
Manufacturing_businesses = in_Lagos_or_Kano[in_Lagos_or_Kano["business_type"] == "Manufacturing"]

Manufacturing_businesses

business_id business_name  state  business_type  start_date  \
22      NG_0023   Business_23   Kano  Manufacturing  2023-01-20   
42      NG_0043   Business_43   Kano  Manufacturing  2022-09-19   
70      NG_0071   Business_71   Kano  Manufacturing  2022-03-01   
75      NG_0076   Business_76   Kano  Manufacturing  2021-09-25   
87      NG_0088   Business_88  Lagos  Manufacturing  2021-08-03   
92      NG_0093   Business_93   Kano  Manufacturing  2020-06-02   
105     NG_0106  Business_106   Kano  Manufacturing  2021-03-28   
122     NG_0123  Business_123   Kano  Manufacturing  2022-01-31   
127     NG_0128  Business_128  Lagos  Manufacturing  2021-05-30   
165     NG_0166  Business_166  Lagos  Manufacturing  2023-01-03   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
22                  589101               332379         11        False   
42                  279155               216038          4        False   
70                  435325               336302          8        False   
75                  419289               335431          6        False   
87                  746502               157898         14        False   
92                  533285               147207          8        False   
105                 414793                82169          6        False   
122                 572033               141804          5        False   
127                 123223                98578          6        False   
165                 288084               230467         11         True   

     accepts_mobile_money  
22                  False  
42                   True  
70                   True  
75                  False  
87                  False  
92                   True  
105                  True  
122                  True  
127                  True  
165                 False

In [25]:
int(Manufacturing_businesses["business_name"].count())

10

There are 10 manufacturing businesses in Lagos or Kano.

### Q4: Small Businesses Analysis
Filter businesses that have fewer than 5 employees AND monthly costs less than 200,000 Naira. What business types do these small businesses represent?

**Business Context:** Small businesses often need different support strategies than larger enterprises.


In [26]:
Fewer_than_5_employees = Nigerian_businesses[(Nigerian_businesses["employees"] < 5 )& (Nigerian_businesses["monthly_costs_naira"] < 200000)]

Fewer_than_5_employees

business_id business_name    state  business_type  start_date  \
16      NG_0017   Business_17   Kaduna         Retail  2021-05-26   
23      NG_0024   Business_24  Anambra       Services  2023-10-09   
30      NG_0031   Business_31      Oyo       Services  2023-02-22   
57      NG_0058   Business_58    Lagos       Services  2022-07-24   
71      NG_0072   Business_72     Kano       Services  2023-02-24   
83      NG_0084   Business_84  Anambra       Services  2022-06-15   
89      NG_0090   Business_90     Kano    Agriculture  2022-05-10   
123     NG_0124  Business_124    Benue         Retail  2021-03-17   
138     NG_0139  Business_139    Benue     Technology  2022-08-14   
153     NG_0154  Business_154   Rivers       Services  2021-12-27   
156     NG_0157  Business_157    Benue    Agriculture  2022-02-14   
181     NG_0182  Business_182    Benue  Manufacturing  2022-10-05   
198     NG_0199  Business_199      Oyo    Agriculture  2022-11-08   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
16                  212107               169685          2         True   
23                  207648               166118          4        False   
30                  795650                40562          3        False   
57                  306577                43908          4         True   
71                  167976                70428          1        False   
83                  805420               163626          4        False   
89                  403124               -29477          4        False   
123                 843719               190578          3        False   
138                  65023                52018          4        False   
153                 239679                13311          4        False   
156                  70828                56662          4         True   
181                 454650                88372          3        False   
198                  98486                78788          4        False   

     accepts_mobile_money  
16                  False  
23                   True  
30                   True  
57                   True  
71                   True  
83                   True  
89                   True  
123                  True  
138                 False  
153                  True  
156                 False  
181                  True  
198                  True

In [27]:
Fewer_than_5_employees['business_type'].unique()

array(['Retail', 'Services', 'Agriculture', 'Technology', 'Manufacturing'],
      dtype=object)

The business types represnted by businesses that have fewer than 5 employees AND monthly costs less than 200,000 Naira are retail, service, agriculture, technology and manufacturing.

### Q5: Profitable Retail Businesses
Filter Retail businesses where monthly revenue is greater than monthly costs. Display their state and employee count.

**Business Context:** Understanding profitable retail operations can help identify successful business practices.


In [28]:
Retail_businesses = Nigerian_businesses[(Nigerian_businesses["business_type"] == "Retail") & (Nigerian_businesses["monthly_revenue_naira"] > Nigerian_businesses["monthly_costs_naira"])]

Retail_businesses.head()

business_id business_name    state business_type  start_date  \
0      NG_0001    Business_1    Benue        Retail  2023-02-01   
16     NG_0017   Business_17   Kaduna        Retail  2021-05-26   
26     NG_0027   Business_27      Imo        Retail  2022-05-03   
27     NG_0028   Business_28     Kano        Retail  2020-02-20   
28     NG_0029   Business_29  Anambra        Retail  2021-07-28   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                  417741               406718          8        False   
16                 212107               169685          2         True   
26                 151342               121073         12        False   
27                 713895               168645         10        False   
28                 467226               270305          3         True   

    accepts_mobile_money  
0                   True  
16                 False  
26                  True  
27                 False  
28                  True

In [29]:
Retail_businesses[["business_name", "state", "employees"]].head()

business_name    state  employees
0     Business_1    Benue          8
16   Business_17   Kaduna          2
26   Business_27      Imo         12
27   Business_28     Kano         10
28   Business_29  Anambra          3

### Q6: Businesses Accepting Mobile Money
Filter all businesses that accept mobile money AND have more than 8 employees. Which states are these businesses located in?

**Business Context:** Larger businesses adopting mobile payments indicate financial technology integration.


In [30]:
Accept_mobile_money = Nigerian_businesses.loc[(Nigerian_businesses["accepts_mobile_money"] == True) & (Nigerian_businesses["employees"] > 8)]

Accept_mobile_money.head()

business_id business_name   state  business_type  start_date  \
2      NG_0003    Business_3     Oyo     Technology  2020-09-02   
6      NG_0007    Business_7     Oyo       Services  2021-06-21   
10     NG_0011   Business_11  Rivers       Services  2022-04-07   
13     NG_0014   Business_14  Rivers    Agriculture  2021-11-15   
21     NG_0022   Business_22     Oyo  Manufacturing  2023-03-12   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
2                  520713                87719         11        False   
6                  724708               207436         12        False   
10                 564903               326094         16        False   
13                 599212               367711         13        False   
21                 264774               211819          9         True   

    accepts_mobile_money  
2                   True  
6                   True  
10                  True  
13                  True  
21                  True

In [31]:
Accept_mobile_money[["business_name","state"]].head()

business_name   state
2     Business_3     Oyo
6     Business_7     Oyo
10   Business_11  Rivers
13   Business_14  Rivers
21   Business_22     Oyo

In [32]:
Accept_mobile_money["state"].unique()

array(['Oyo', 'Rivers', 'Imo', 'Kaduna', 'Lagos', 'Benue', 'Anambra',
       'Kano'], dtype=object)

All businesses that accept mobile money AND have more than 8 employees are located in Oyo, Rivers, Imo, Kaduna, Lagos, Benue, Anambra and Kano.

### Q7: Low-Cost Operations
Filter businesses where monthly costs are less than 150,000 Naira. What is the most common business type among these low-cost operations?

**Business Context:** Low-cost business models can be important for economic inclusion and job creation.


In [33]:
less_than_150k = Nigerian_businesses[Nigerian_businesses["monthly_costs_naira"] < 150000]

less_than_150k.head()

business_id business_name    state business_type  start_date  \
2      NG_0003    Business_3      Oyo    Technology  2020-09-02   
4      NG_0005    Business_5   Rivers    Technology  2022-11-28   
5      NG_0006    Business_6  Anambra    Technology  2022-01-28   
7      NG_0008    Business_8      Oyo    Technology  2022-07-09   
26     NG_0027   Business_27      Imo        Retail  2022-05-03   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
2                  520713                87719         11        False   
4                  231537               -11869         10         True   
5                 -172819              -138255          5        False   
7                  928641               111173         11        False   
26                 151342               121073         12        False   

    accepts_mobile_money  
2                   True  
4                  False  
5                  False  
7                  False  
26                  True

In [34]:
less_than_150k["business_type"].value_counts()

business_type
Retail           12
Services         11
Technology        9
Manufacturing     8
Agriculture       7
Name: count, dtype: int64

The most common business type smong the low cost operations is the Retail business type.

### Q8: Recent Business Registrations
Filter businesses that started in 2023. Which business type is most common among these new businesses?

**Business Context:** Recent registration trends can indicate emerging business opportunities and economic growth sectors.


In [35]:
started_in_2023 = Nigerian_businesses[Nigerian_businesses["start_date"].between("2023-01-01","2023-12-31")]

started_in_2023.head()

business_id business_name   state  business_type  start_date  \
0      NG_0001    Business_1   Benue         Retail  2023-02-01   
1      NG_0002    Business_2  Kaduna  Manufacturing  2023-11-17   
3      NG_0004    Business_4   Benue  Manufacturing  2023-04-14   
21     NG_0022   Business_22     Oyo  Manufacturing  2023-03-12   
22     NG_0023   Business_23    Kano  Manufacturing  2023-01-20   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                  417741               406718          8        False   
1                  696804               256593         10         True   
3                  553527               389310          7        False   
21                 264774               211819          9         True   
22                 589101               332379         11        False   

    accepts_mobile_money  
0                   True  
1                  False  
3                   True  
21                  True  
22                 False

In [36]:
started_in_2023["business_type"].value_counts()

business_type
Retail           17
Manufacturing    14
Services          9
Technology        7
Agriculture       3
Name: count, dtype: int64

Most new businesses started in 2023 are Retail businesses

### Q9: Large Employers
Filter businesses that have 10 or more employees. Calculate the average monthly revenue of these larger employers.

**Business Context:** Large employers are important for job creation and economic stability.


In [37]:
more_than_10_employees = Nigerian_businesses[Nigerian_businesses["employees"] >= 10]

more_than_10_employees.head()

business_id business_name   state  business_type  start_date  \
1     NG_0002    Business_2  Kaduna  Manufacturing  2023-11-17   
2     NG_0003    Business_3     Oyo     Technology  2020-09-02   
4     NG_0005    Business_5  Rivers     Technology  2022-11-28   
6     NG_0007    Business_7     Oyo       Services  2021-06-21   
7     NG_0008    Business_8     Oyo     Technology  2022-07-09   

   monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
1                 696804               256593         10         True   
2                 520713                87719         11        False   
4                 231537               -11869         10         True   
6                 724708               207436         12        False   
7                 928641               111173         11        False   

   accepts_mobile_money  
1                 False  
2                  True  
4                 False  
6                  True  
7                 False

In [38]:
float(more_than_10_employees["monthly_revenue_naira"].mean())

511441.8548387097

The average monthly revenue of businesses that have 10 or more employees is approximately 511,442.

### Q10: Digital-Ready Businesses
Filter businesses that have a website AND accept mobile money. What percentage of the total businesses are digitally ready?

**Business Context:** Digital readiness indicates how prepared businesses are for the modern economy.


In [39]:
Digitally_ready = Nigerian_businesses.loc[(Nigerian_businesses["accepts_mobile_money"] == True) & (Nigerian_businesses["has_website"] == True)]

Digitally_ready.head()

business_id business_name    state  business_type  start_date  \
14     NG_0015   Business_15  Anambra    Agriculture  2020-01-20   
21     NG_0022   Business_22      Oyo  Manufacturing  2023-03-12   
28     NG_0029   Business_29  Anambra         Retail  2021-07-28   
33     NG_0034   Business_34     Kano     Technology  2021-07-23   
34     NG_0035   Business_35      Imo         Retail  2023-09-27   

    monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
14                 511114               422377          8         True   
21                 264774               211819          9         True   
28                 467226               270305          3         True   
33                 265604                 8259          8         True   
34                 182707                89967          5         True   

    accepts_mobile_money  
14                  True  
21                  True  
28                  True  
33                  True  
34                  True

In [40]:
int(Digitally_ready["business_name"].count() / Nigerian_businesses["business_name"].count() * 100)

14

14% of the total businesses are digitally ready

# Exercise 2: African Agriculture Analysis

**AgriTech Africa** is a research organization analyzing agricultural technology adoption across five African countries. You have been provided with a dataset containing information about farming communities and their technology usage patterns.

For each question below, provide the Python pandas code and provide a key insight you can get from your answer.


In [41]:
AgriTech_Africa = pd.read_csv("data/african_agriculture_dataset.csv")

AgriTech_Africa.head(10)

Community_Name       Country Crop_Type  Farm_Size  \
0         Mountain Cultivators Club  South Africa    Barley      39.96   
1          Prairie Cultivators Club  South Africa   Sorghum      77.62   
2              Prairie Agri Society       Nigeria     Wheat      97.46   
3             Harvest Farmers Union       Nigeria     Maize      27.84   
4              Valley Farmers Group         Ghana   Sorghum      70.60   
5             Savannah Agri Society      Tanzania    Barley      92.92   
6           River Growers Coalition  South Africa   Sorghum      68.94   
7            Savannah Farmers Union       Nigeria      Rice      68.43   
8  Prairie Agricultural Association      Tanzania     Wheat      23.15   
9         Sunrise Growers Coalition         Kenya     Maize      53.98   

   Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield Fertilizer_Type  \
0                   40         Yes          SMS        318.05         Organic   
1                   48          No          SMS        704.85         Organic   
2                   14         Yes          SMS        439.41             NaN   
3                   55          No          SMS        668.23             NaN   
4                   53         Yes  Mobile Apps        801.16             NaN   
5                   30         Yes          SMS        784.90         Organic   
6                   60          No          NaN        437.78       Inorganic   
7                   57          No          NaN        611.11         Organic   
8                   86         Yes          SMS        504.69         Organic   
9                   59         Yes          NaN        883.72         Organic   

  Irrigation Weather_Station  Annual_Rainfall  
0  Sprinkler             Yes          1046.21  
1       Drip              No           921.82  
2       Drip             Yes          1072.58  
3  Sprinkler              No           629.73  
4  Sprinkler              No           909.69  
5  Sprinkler              No           872.48  
6        NaN             Yes           897.94  
7       Drip              No           534.91  
8       Drip             Yes           621.79  
9        NaN              No           629.33

In [42]:
AgriTech_Africa.describe()

Farm_Size  Tech_Adoption_Index  Annual_Yield  Annual_Rainfall
count  2000.000000          2000.000000   2000.000000      2000.000000
mean     52.126350            50.044500    536.689350       794.925925
std      28.786657            28.947133    257.207925       228.165699
min       1.000000             0.000000    100.110000       400.000000
25%      27.330000            25.000000    311.357500       596.695000
50%      53.675000            50.000000    536.050000       798.695000
75%      76.907500            75.000000    752.900000       983.485000
max      99.970000           100.000000    999.730000      1199.830000

## Q1: Identifying Technology Leaders
**Business Question:** Which farming communities are leading in technology adoption, and how many of our partner communities have achieved a Technology Adoption Index above 80?

**Context:** Understanding technology leaders helps identify best practices for scaling across other communities.

In [43]:
leading_tech_adoption = AgriTech_Africa.sort_values("Tech_Adoption_Index", ascending=False)

leading_tech_adoption.head(10)

Community_Name       Country Crop_Type  Farm_Size  \
698   Golden Growers Coalition       Nigeria    Barley      77.71   
671      Harvest Farmers Union         Ghana   Sorghum      59.91   
612      Sunrise Farmers Group      Tanzania     Wheat       7.26   
1256        Green Agri Society  South Africa     Wheat      54.27   
1305    Savannah Farmers Union  South Africa    Barley      42.58   
225      Mountain Agri Society      Tanzania      Rice      84.17   
931        River Farmers Group      Tanzania   Sorghum      45.19   
1430      Meadow Farmers Group      Tanzania     Wheat      34.21   
731        Valley Agri Society         Ghana     Maize      79.25   
1611   River Growers Coalition         Ghana      Rice      98.30   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
698                   100         Yes  Mobile Apps        387.09   
671                   100         Yes  Mobile Apps        597.05   
612                   100         Yes  Mobile Apps        754.81   
1256                  100         Yes          NaN        264.99   
1305                  100         Yes          SMS        184.80   
225                   100         Yes          SMS        411.85   
931                   100         Yes          SMS        452.77   
1430                  100          No          SMS        214.97   
731                   100          No  Mobile Apps        814.08   
1611                  100         Yes          NaN        105.33   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
698        Inorganic  Sprinkler             Yes           477.69  
671        Inorganic        NaN             Yes           841.94  
612              NaN       Drip             Yes           712.89  
1256             NaN        NaN             Yes           519.53  
1305       Inorganic  Sprinkler              No           461.35  
225        Inorganic       Drip              No           793.30  
931              NaN  Sprinkler             Yes           882.55  
1430             NaN  Sprinkler              No           962.02  
731              NaN        NaN             Yes           851.31  
1611         Organic  Sprinkler             Yes           853.19

In [44]:
above_80 = AgriTech_Africa[AgriTech_Africa["Tech_Adoption_Index"] >= 80]

above_80.head()

Community_Name       Country Crop_Type  Farm_Size  \
8   Prairie Agricultural Association      Tanzania     Wheat      23.15   
11         Sunrise Growers Coalition  South Africa     Wheat      70.31   
16                River Agri Society      Tanzania   Sorghum      74.55   
17             Savannah Agri Society  South Africa   Sorghum      14.50   
20         Savannah Cultivators Club         Kenya   Sorghum      46.27   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
8                    86         Yes          SMS        504.69   
11                   85         Yes          SMS        409.37   
16                   83          No  Mobile Apps        691.11   
17                   91          No  Mobile Apps        398.23   
20                   92          No          SMS        686.81   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
8          Organic       Drip             Yes           621.79  
11         Organic  Sprinkler              No          1178.04  
16       Inorganic  Sprinkler             Yes           719.05  
17       Inorganic       Drip             Yes          1067.74  
20             NaN        NaN             Yes           612.64

In [45]:
int(above_80["Community_Name"].count())

409

409 of our partner communities have achieved a Technology Adoption Index above 80?

## Q2: Premium Drone Adoption Analysis
**Business Question:** Among large-scale farming operations (over 70 acres), which communities have invested in drone technology, and where are they located?

**Context:** Large farms with drone adoption represent high-value clients for advanced agricultural technology partnerships.

In [46]:
large_scale_farmers = AgriTech_Africa[AgriTech_Africa["Farm_Size"] > 70]

large_scale_farmers.head()

Community_Name       Country Crop_Type  Farm_Size  \
1    Prairie Cultivators Club  South Africa   Sorghum      77.62   
2        Prairie Agri Society       Nigeria     Wheat      97.46   
4        Valley Farmers Group         Ghana   Sorghum      70.60   
5       Savannah Agri Society      Tanzania    Barley      92.92   
11  Sunrise Growers Coalition  South Africa     Wheat      70.31   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
1                    48          No          SMS        704.85   
2                    14         Yes          SMS        439.41   
4                    53         Yes  Mobile Apps        801.16   
5                    30         Yes          SMS        784.90   
11                   85         Yes          SMS        409.37   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
1          Organic       Drip              No           921.82  
2              NaN       Drip             Yes          1072.58  
4              NaN  Sprinkler              No           909.69  
5          Organic  Sprinkler              No           872.48  
11         Organic  Sprinkler              No          1178.04

In [47]:
drone_tech_farmers = large_scale_farmers[large_scale_farmers["Drone_Usage"] == "Yes"]

drone_tech_farmers.head()

Community_Name       Country Crop_Type  Farm_Size  \
2        Prairie Agri Society       Nigeria     Wheat      97.46   
4        Valley Farmers Group         Ghana   Sorghum      70.60   
5       Savannah Agri Society      Tanzania    Barley      92.92   
11  Sunrise Growers Coalition  South Africa     Wheat      70.31   
24       Meadow Farmers Group         Ghana      Rice      94.51   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
2                    14         Yes          SMS        439.41   
4                    53         Yes  Mobile Apps        801.16   
5                    30         Yes          SMS        784.90   
11                   85         Yes          SMS        409.37   
24                   51         Yes          NaN        883.63   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
2              NaN       Drip             Yes          1072.58  
4              NaN  Sprinkler              No           909.69  
5          Organic  Sprinkler              No           872.48  
11         Organic  Sprinkler              No          1178.04  
24             NaN        NaN              No           917.49

In [48]:
drone_tech_farmers[["Community_Name", "Country"]].head()

Community_Name       Country
2        Prairie Agri Society       Nigeria
4        Valley Farmers Group         Ghana
5       Savannah Agri Society      Tanzania
11  Sunrise Growers Coalition  South Africa
24       Meadow Farmers Group         Ghana

In [49]:
drone_tech_farmers["Country"].unique()

array(['Nigeria', 'Ghana', 'Tanzania', 'South Africa', 'Kenya'],
      dtype=object)

Among large-scale farming operations (over 70 acres), the communities that have invested in drone technology, are located in Nigeria, Ghana, Tanzania, South Africa and Kenya.

## Q3: Market Expansion Opportunities
**Business Question:** How many farming communities are operating in our priority markets of South Africa and Kenya?

**Context:** These countries represent key expansion opportunities for agricultural technology solutions.

In [50]:
In_priority_markets = AgriTech_Africa[(AgriTech_Africa["Country"] == "South Africa") | (AgriTech_Africa["Country"] == "Kenya")]

In_priority_markets.head()

Community_Name       Country Crop_Type  Farm_Size  \
0   Mountain Cultivators Club  South Africa    Barley      39.96   
1    Prairie Cultivators Club  South Africa   Sorghum      77.62   
6     River Growers Coalition  South Africa   Sorghum      68.94   
9   Sunrise Growers Coalition         Kenya     Maize      53.98   
11  Sunrise Growers Coalition  South Africa     Wheat      70.31   

    Tech_Adoption_Index Drone_Usage Mobile_Tech  Annual_Yield Fertilizer_Type  \
0                    40         Yes         SMS        318.05         Organic   
1                    48          No         SMS        704.85         Organic   
6                    60          No         NaN        437.78       Inorganic   
9                    59         Yes         NaN        883.72         Organic   
11                   85         Yes         SMS        409.37         Organic   

   Irrigation Weather_Station  Annual_Rainfall  
0   Sprinkler             Yes          1046.21  
1        Drip              No           921.82  
6         NaN             Yes           897.94  
9         NaN              No           629.33  
11  Sprinkler              No          1178.04

In [51]:
int(In_priority_markets["Community_Name"].count())

760

760 farming communities are operating in our priority markets of South Africa and Kenya.

## Q4: Underserved Market Identification
**Business Question:** Which countries have small-scale, low-technology farming operations (under 20 acres with Technology Adoption Index below 30) that could benefit from basic technology interventions?

**Context:** Small, low-tech farms represent an underserved market segment requiring tailored entry-level solutions.


In [52]:
small_scale = AgriTech_Africa[(AgriTech_Africa["Farm_Size"] < 20) & (AgriTech_Africa["Tech_Adoption_Index"] < 30)]

small_scale.head()

Community_Name       Country Crop_Type  Farm_Size  \
15               Meadow Agri Society       Nigeria    Barley       8.72   
41   Valley Agricultural Association       Nigeria      Rice       2.16   
53            River Cultivators Club  South Africa    Barley       9.59   
54  Sunrise Agricultural Association  South Africa     Wheat       3.14   
81        Mountain Growers Coalition         Kenya      Rice      13.85   

    Tech_Adoption_Index Drone_Usage Mobile_Tech  Annual_Yield Fertilizer_Type  \
15                   28          No         SMS        607.82         Organic   
41                   28         Yes         SMS        232.53             NaN   
53                   28         Yes         SMS        255.60         Organic   
54                   10          No         NaN        531.49       Inorganic   
81                    1          No         NaN        801.18             NaN   

   Irrigation Weather_Station  Annual_Rainfall  
15       Drip             Yes           428.85  
41  Sprinkler              No           522.66  
53        NaN              No           813.43  
54  Sprinkler             Yes           405.82  
81  Sprinkler              No          1007.81

In [53]:
small_scale["Country"].unique()

array(['Nigeria', 'South Africa', 'Kenya', 'Tanzania', 'Ghana'],
      dtype=object)

The countries that have small scale and low technology farming are Nigeria, South Africa, Kenya, Tanzania and Ghana

## Q5: Sustainable Agriculture Success Stories
**Business Question:** Which organic farming communities are achieving high productivity (over 600 annual yield), and what irrigation technologies are they using?

**Context:** Successful organic farms demonstrate sustainable practices that can be promoted to environmentally conscious markets.

In [54]:
productive_farms = AgriTech_Africa[AgriTech_Africa["Annual_Yield"] > 600]

productive_farms.head()

Community_Name       Country Crop_Type  Farm_Size  \
1  Prairie Cultivators Club  South Africa   Sorghum      77.62   
3     Harvest Farmers Union       Nigeria     Maize      27.84   
4      Valley Farmers Group         Ghana   Sorghum      70.60   
5     Savannah Agri Society      Tanzania    Barley      92.92   
7    Savannah Farmers Union       Nigeria      Rice      68.43   

   Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield Fertilizer_Type  \
1                   48          No          SMS        704.85         Organic   
3                   55          No          SMS        668.23             NaN   
4                   53         Yes  Mobile Apps        801.16             NaN   
5                   30         Yes          SMS        784.90         Organic   
7                   57          No          NaN        611.11         Organic   

  Irrigation Weather_Station  Annual_Rainfall  
1       Drip              No           921.82  
3  Sprinkler              No           629.73  
4  Sprinkler              No           909.69  
5  Sprinkler              No           872.48  
7       Drip              No           534.91

In [55]:
productive_farms[["Community_Name", "Irrigation"]].head()

Community_Name Irrigation
1  Prairie Cultivators Club       Drip
3     Harvest Farmers Union  Sprinkler
4      Valley Farmers Group  Sprinkler
5     Savannah Agri Society  Sprinkler
7    Savannah Farmers Union       Drip

In [56]:
productive_farms["Irrigation"].unique()

array(['Drip', 'Sprinkler', nan], dtype=object)

Some organic farming communities achieving high productivity (over 600 annual yield) use drip or sprinkler irrigation technologies. 

## Q6: Climate-Smart Agriculture Adoption
**Business Question:** In high-rainfall regions (over 1000mm annually), which countries have farming communities that have invested in weather monitoring technology?

**Context:** Weather station adoption in high-rainfall areas indicates sophisticated climate risk management practices.


In [57]:
Countries_with_high_rainfall = AgriTech_Africa[AgriTech_Africa["Annual_Rainfall"] > 1000]

Countries_with_high_rainfall.head()

Community_Name       Country Crop_Type  Farm_Size  \
0         Mountain Cultivators Club  South Africa    Barley      39.96   
2              Prairie Agri Society       Nigeria     Wheat      97.46   
11        Sunrise Growers Coalition  South Africa     Wheat      70.31   
12  Valley Agricultural Association         Ghana    Barley      99.69   
17            Savannah Agri Society  South Africa   Sorghum      14.50   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
0                    40         Yes          SMS        318.05   
2                    14         Yes          SMS        439.41   
11                   85         Yes          SMS        409.37   
12                   60          No  Mobile Apps        182.56   
17                   91          No  Mobile Apps        398.23   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
0          Organic  Sprinkler             Yes          1046.21  
2              NaN       Drip             Yes          1072.58  
11         Organic  Sprinkler              No          1178.04  
12         Organic       Drip             Yes          1163.25  
17       Inorganic       Drip             Yes          1067.74

In [58]:
have_weather_monitoring_tech = Countries_with_high_rainfall[Countries_with_high_rainfall["Weather_Station"] == "Yes"]

have_weather_monitoring_tech[["Community_Name", "Country", "Annual_Rainfall", "Weather_Station"]].head()

Community_Name       Country  Annual_Rainfall  \
0         Mountain Cultivators Club  South Africa          1046.21   
2              Prairie Agri Society       Nigeria          1072.58   
12  Valley Agricultural Association         Ghana          1163.25   
17            Savannah Agri Society  South Africa          1067.74   
18            Mountain Agri Society       Nigeria          1132.60   

   Weather_Station  
0              Yes  
2              Yes  
12             Yes  
17             Yes  
18             Yes

In [59]:
have_weather_monitoring_tech["Country"].unique()

array(['South Africa', 'Nigeria', 'Ghana', 'Tanzania', 'Kenya'],
      dtype=object)

In high-rainfall regions (over 1000mm annually), the countries that have farming communities that have invested in weather monitoring technology are South Africa, Nigeria, Ghana, Tanzania and Kenya.

## Q7: Entry-Level Market Assessment
**Business Question:** Among low-performing farming operations (annual yield under 200), which crop types are most prevalent and may need immediate intervention?

**Context:** Low-yield operations represent potential clients for productivity improvement solutions.

In [60]:
low_performing_farms = AgriTech_Africa[AgriTech_Africa["Annual_Yield"] < 200]

low_performing_farms.head()

Community_Name  Country Crop_Type  Farm_Size  \
12  Valley Agricultural Association    Ghana    Barley      99.69   
30            Harvest Farmers Union  Nigeria   Sorghum       5.88   
39  Meadow Agricultural Association    Ghana      Rice      16.93   
44           River Cultivators Club    Ghana    Barley      31.74   
55            Mountain Agri Society  Nigeria     Wheat      59.81   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
12                   60          No  Mobile Apps        182.56   
30                   83         Yes  Mobile Apps        136.35   
39                   71          No          NaN        181.94   
44                   46         Yes  Mobile Apps        198.41   
55                   52          No          SMS        154.01   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
12         Organic       Drip             Yes          1163.25  
30             NaN       Drip             Yes          1024.79  
39         Organic        NaN             Yes          1052.89  
44         Organic       Drip             Yes           498.94  
55         Organic       Drip             Yes          1137.86

In [61]:
low_performing_farms["Crop_Type"].unique()

array(['Barley', 'Sorghum', 'Rice', 'Wheat', 'Maize'], dtype=object)

Among low performing farming operations, barley, sorghum, rice, wheat and maize are the most prevalent crop types that may need immediate intervention.

## Q8: Digital Agriculture Penetration
**Business Question:** What irrigation methods are most commonly used by digitally-enabled farming communities (those using Mobile Apps or SMS technology)?

**Context:** Understanding technology adoption patterns helps design integrated digital agriculture solutions.

In [62]:
digitally_enabled_farming = AgriTech_Africa[(AgriTech_Africa["Mobile_Tech"] == "Mobile Apps") | (AgriTech_Africa["Mobile_Tech"] == "SMS")]

digitally_enabled_farming.head()

Community_Name       Country Crop_Type  Farm_Size  \
0  Mountain Cultivators Club  South Africa    Barley      39.96   
1   Prairie Cultivators Club  South Africa   Sorghum      77.62   
2       Prairie Agri Society       Nigeria     Wheat      97.46   
3      Harvest Farmers Union       Nigeria     Maize      27.84   
4       Valley Farmers Group         Ghana   Sorghum      70.60   

   Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield Fertilizer_Type  \
0                   40         Yes          SMS        318.05         Organic   
1                   48          No          SMS        704.85         Organic   
2                   14         Yes          SMS        439.41             NaN   
3                   55          No          SMS        668.23             NaN   
4                   53         Yes  Mobile Apps        801.16             NaN   

  Irrigation Weather_Station  Annual_Rainfall  
0  Sprinkler             Yes          1046.21  
1       Drip              No           921.82  
2       Drip             Yes          1072.58  
3  Sprinkler              No           629.73  
4  Sprinkler              No           909.69

In [63]:
digitally_enabled_farming["Irrigation"].unique()

array(['Sprinkler', 'Drip', nan], dtype=object)

The most commonly used irrigation methods by digitally-enabled farming communities are Springler and Drip. 

## Q9: Large-Scale Technology Investment Patterns
**Business Question:** What is the average technology adoption level among large-scale farming operations (90+ acres), and how does it compare to smaller operations?

**Context:** Large farms typically have higher investment capacity and can be early adopters of expensive technologies.

In [64]:
large_scale = AgriTech_Africa[AgriTech_Africa["Farm_Size"] >= 90]

large_scale.head()

Community_Name   Country Crop_Type  Farm_Size  \
2              Prairie Agri Society   Nigeria     Wheat      97.46   
5             Savannah Agri Society  Tanzania    Barley      92.92   
12  Valley Agricultural Association     Ghana    Barley      99.69   
24             Meadow Farmers Group     Ghana      Rice      94.51   
28  Valley Agricultural Association   Nigeria     Maize      98.92   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
2                    14         Yes          SMS        439.41   
5                    30         Yes          SMS        784.90   
12                   60          No  Mobile Apps        182.56   
24                   51         Yes          NaN        883.63   
28                    1          No          NaN        571.98   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
2              NaN       Drip             Yes          1072.58  
5          Organic  Sprinkler              No           872.48  
12         Organic       Drip             Yes          1163.25  
24             NaN        NaN              No           917.49  
28             NaN  Sprinkler              No           582.40

In [65]:
float(large_scale["Tech_Adoption_Index"].mean())

49.242009132420094

In [66]:
float(small_scale["Tech_Adoption_Index"].mean())

14.274336283185841

The average technology adoption level among large-scale farming operations is approximately 49 which is relatively higher than the average technology adoption level among small-scale farming operations which is approximately 14.


## Q10: Natural Farming Excellence
**Business Question:** What percentage of farming communities achieve high productivity (over 500 annual yield) without using fertilizers, demonstrating natural farming excellence?

**Context:** High-performing farms without fertilizer inputs represent efficient operations that could mentor others in sustainable practices.


In [67]:
highly_productive = AgriTech_Africa[AgriTech_Africa["Annual_Yield"] > 500]

highly_productive.head()

Community_Name       Country Crop_Type  Farm_Size  \
1  Prairie Cultivators Club  South Africa   Sorghum      77.62   
3     Harvest Farmers Union       Nigeria     Maize      27.84   
4      Valley Farmers Group         Ghana   Sorghum      70.60   
5     Savannah Agri Society      Tanzania    Barley      92.92   
7    Savannah Farmers Union       Nigeria      Rice      68.43   

   Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield Fertilizer_Type  \
1                   48          No          SMS        704.85         Organic   
3                   55          No          SMS        668.23             NaN   
4                   53         Yes  Mobile Apps        801.16             NaN   
5                   30         Yes          SMS        784.90         Organic   
7                   57          No          NaN        611.11         Organic   

  Irrigation Weather_Station  Annual_Rainfall  
1       Drip              No           921.82  
3  Sprinkler              No           629.73  
4  Sprinkler              No           909.69  
5  Sprinkler              No           872.48  
7       Drip              No           534.91

In [68]:
natural_farms = highly_productive[highly_productive["Fertilizer_Type"].isnull()]

natural_farms.head()

Community_Name  Country Crop_Type  Farm_Size  \
3       Harvest Farmers Union  Nigeria     Maize      27.84   
4        Valley Farmers Group    Ghana   Sorghum      70.60   
18      Mountain Agri Society  Nigeria     Wheat      62.62   
19       Valley Farmers Group    Kenya     Maize      33.14   
20  Savannah Cultivators Club    Kenya   Sorghum      46.27   

    Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
3                    55          No          SMS        668.23   
4                    53         Yes  Mobile Apps        801.16   
18                   26         Yes          NaN        700.75   
19                   43          No  Mobile Apps        943.97   
20                   92          No          SMS        686.81   

   Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
3              NaN  Sprinkler              No           629.73  
4              NaN  Sprinkler              No           909.69  
18             NaN  Sprinkler             Yes          1132.60  
19             NaN        NaN              No           589.20  
20             NaN        NaN             Yes           612.64

In [69]:
float(natural_farms["Community_Name"].count() / AgriTech_Africa["Community_Name"].count() * 100)

18.7

18.7% of farming communities achieve high productivity (over 500 annual yield) without using fertilizers, demonstrating natural farming excellence